lightning/src/ln/chan_utils.rs

```diff
@@ -562,7 +563,7 @@ impl LocalCommitmentTransaction {
 
 	/// Generate a new LocalCommitmentTransaction based on a raw commitment transaction,
 	/// remote signature and both parties keys
-	pub(crate) fn new_missing_local_sig(unsigned_tx: Transaction, their_sig: Signature, our_funding_key: &PublicKey, their_funding_key: &PublicKey, local_keys: TxCreationKeys, feerate_per_kw: u32, htlc_data: Vec<(HTLCOutputInCommitment, Option<Signature>)>) -> LocalCommitmentTransaction {
+	pub fn new_missing_local_sig(unsigned_tx: Transaction, their_sig: Signature, our_funding_key: &PublicKey, their_funding_key: &PublicKey, local_keys: TxCreationKeys, feerate_per_kw: u32, htlc_data: Vec<(HTLCOutputInCommitment, Option<Signature>)>) -> LocalCommitmentTransaction {
```

Comment:
What use do you have for this?

Reply:
Need to be able to construct a `LocalCommitmentTransaction` to pass into `ChannelKeys.sign_local_commitment`.  So it's a similar reason to allowing creation of `TxCreationKeys`.

Reply:
If we're gonna make this pub, it should probably be more strict and we should document what the input must look like. eg we should check the outputs in the unsigned_tx match those passed in htlc_data and we should document the cases that panic.

Reply:
Are you thinking we should just check that the outputs are present, or that the whole `TxOut` matches?

Reply:
Ideally that the whole TxOut matches, but if it requires hashing or something expensive, then feel free to skip it. Up to you.

Reply:
Well, there's definitely a hash due to p2wsh.

Reply:
Took a pass at this, take a look